src/plugin.ts

```diff
@@ -643,6 +639,11 @@ export class MaestroPlugin implements CompilerPlugin {
             }
             console.timeEnd('Inject bindings into xml files');
         }
+
+        //do some nodeclass transformations
+        for (let nc of Object.values(this.fileMap.nodeClasses)) {

```

Comment:
Moving this loop out of `beforeFileTranspile` (which runs once for _every_ file in the program per transpile cycle) and into `beforeProgramTranspile` (run _once_ per transpile cycle) should yield some significant performance improvements.